src/adapters/github.js

```diff
@@ -342,16 +347,22 @@ class GitHub extends PjaxAdapter {
           const hugeRepos = this.store.get(STORE.HUGE_REPOS);
           const repo = `${opts.repo.username}/${opts.repo.reponame}`;
           const repos = Object.keys(hugeRepos);
-          if (!hugeRepos[repo]) {
-            // If there are too many repos memoized, delete the oldest one
-            if (repos.length >= GH_MAX_HUGE_REPOS_SIZE) {
-              const oldestRepo = repos.reduce((min, p) => (hugeRepos[p] < hugeRepos[min] ? p : min));
-              delete hugeRepos[oldestRepo];
-            }
-            hugeRepos[repo] = new Date().getTime();
-            this.store.set(STORE.HUGE_REPOS, hugeRepos);
+
+          const handleHugeRepo = (isFullStorage) => this._handleError(cfg, {status: isFullStorage ? 600 : 206}, cb);
```

Comment:
Why 'handleHugeRepo' calls handleError? Just doesn't make any sense.

Reply:
I will think a better name. Basically, the function just determines whether it should pass the full local storage status or the huge repo status to the `handleError`